### ISSUE TITLE
[Bug] Massive performance drop at edge of ship in March when LLL is enabled

### ISSUE THREAD

**Issue by ajsglist** [IAmBatby/LethalLevelLoader#132]
Hey there,

I've finally managed to track down the mod causing major performance issues at the edge of the ship, and unfortunately it seems to be LLL.  Currently using LLL 1.3.8, and when passing through the ship's load zone, massive performance issues are observed.  When standing in this area, the performance issues persist.  I've captured a few gifs to demonstrate the appreciable performance hit and attached them below.

Vanilla behavior (no mods): full 60fps as expected

https://i.imgur.com/YYtejbG.gif

Modded behavior w/ LLL (poolrooms): performance issues in this area

https://i.imgur.com/n2gQFpz.gif

Modded behavior w/ LLL (mantif): performance issues in this area

https://i.imgur.com/iSY3cbB.gif

Modded behavior w/o LLL (disabled): performance returns to vanilla levels, 60fps as expected

https://i.imgur.com/vb2WtkR.gif

Profile code: 019106ed-a6ce-ccee-d21c-427d61240d4d

Thanks!

**Comment by IAmBatby** [comment]
Initially due to the lack of performance comparisons with vanilla dungeons I'm going to suggest this is not directly due to LethalLevelLoader. In the base game the dungeons lack occlusion culling which may be a problem amplified by this particular area as well as more resource intensive dungeons. Have you tried using the mod Cull Factory to see if this resolves the issue?

**Comment by ajsglist** [comment]
Cull factory is included in the profile, yes.

Disabling LLL removes the performance issue (as demonstrated in the OP).  The only apparent possibilities from the data provided are: LLL itself, LLL loading a problematic interior/moon erroneously (which I attempted to troubleshoot by loading different interiors), or a nasty mod conflict that only manifests on LLL's side (disabling other mods in the profile yielding no benefit)

**Comment by IAmBatby** [comment]
All four tests in the OP imply your performance is fine with vanilla interiors, and not fine with custom interiors, no?

**Comment by ajsglist** [comment]
No, the performance for the custom interiors is fine (again, there is no notable strain on the GPU from either vanilla interiors or custom interiors via LLL).  The performance issue triggers at a reproducible distance from the ship consistent with "reentering" the ship (e.g. the distance at which ship mods traditionally cull themselves).  The 5th gif (see post above) is taken a few feet further from the ship -- this gif also has a custom interior.

**Comment by ajsglist** [comment]
To be clear: the performance issues are also observed when LLL generates the default Facility interior.  Do you have any insight into what's going on here and how it can be fixed?
